packages/flutter_android_lifecycle/example/android/app/src/main/java/io/flutter/plugins/flutter_android_lifecycle_example/MainActivity.java

```diff
@@ -0,0 +1,43 @@
+// Copyright 2019 The Chromium Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+package io.flutter.plugins.flutter_android_lifecycle_example;
+
+import android.util.Log;
+import androidx.lifecycle.Lifecycle;
+import io.flutter.embedding.android.FlutterActivity;
+import io.flutter.embedding.engine.FlutterEngine;
+import io.flutter.embedding.engine.plugins.FlutterPlugin;
+import io.flutter.embedding.engine.plugins.lifecycle.FlutterLifecycleAdapter;
+
+public class MainActivity extends FlutterActivity {
+  private static final String TAG = "MainActivity";
+
+  @Override
+  public void configureFlutterEngine(FlutterEngine flutterEngine) {
+    flutterEngine.getPlugins().add(new TestPlugin());
+  }
+
+  private static class TestPlugin implements FlutterPlugin {
+
+    @Override
+    public void onAttachedToEngine(FlutterPluginBinding flutterPluginBinding) {
+      FlutterLifecycleAdapter adapter =
+          new FlutterLifecycleAdapter(flutterPluginBinding.getLifecycle());
+      Lifecycle lifecycle = adapter.getLifecycle();
+
+      if (lifecycle == null) {
+        Log.d(TAG, "Couldn't obtained Lifecycle!");
+        // TODO(amirh): make this throw once the lifecycle API is available on stable.
+        // throw new RuntimeException(
+        //     "The FlutterLifecycleAdapter did not correctly provide a Lifecycle instance. Source reference: "
+        //         + flutterPluginBinding.getLifecycle());
+      }
+      Log.d(TAG, "Successfully obtained Lifecycle: " + lifecycle);
```

Comment:
nit: This should be behind an `else` if the `throw` is commented out.

Reply:
done

Reply:
(the change was returning early)

Reply:
Whoops, my mistake.